### PR TITLE
ENH add fits readers

### PR DESCRIPTION
**Pull request by beckermr**
This PR adds the fits readers. A bunch of associated changes had to go in as well to make this feasible.

### CODE REVIEW

File: jax_galsim/gsparams.py [hunk 1]
```diff
@@ -21,13 +21,43 @@ class GSParams:
...
 
         In the former case, it returns gsparams, in the latter it returns default
         (GSParams.default if no other default specified).
         """
+        # we allow conversion from the original GalSim GSParams object
+        if isinstance(gsparams, _galsim.GSParams):
+            gsparams = GSParams.from_galsim(gsparams)
+        if default is not None and isinstance(default, _galsim.GSParams):
+            default = GSParams.from_galsim(default)
+
```
Comment (beckermr) [inline jax_galsim/gsparams.py:60]: Sometimes these things get injected during testing. If we'd rather have `check` error on mixing, then I'll need to go back to something else to fix this.

File: tests/GalSim
Comment (beckermr) [inline tests/GalSim:1]: I had to do a bunch of updates to the galsim test suite. I pushed to the same branch that @EiffL was using.

File: tests/jax/galsim/test_image_jax.py
Comment (beckermr) [inline tests/jax/galsim/test_image_jax.py:1]: The changes here will need to be put into galsim.

File: jax_galsim/bounds.py [hunk 2]
```diff
@@ -126,10 +128,12 @@ def includes(self, *args):
         else:
             raise TypeError("include takes at most 2 arguments (%d given)" % len(args))
 
-    def expand(self, factor):
+    def expand(self, factor, yfactor=None):
```
Comment (ismael-mendoza) [inline jax_galsim/bounds.py:131]: any reason to have a different naming convention than the `expand` in the original Galsim? (uses `factor_x` and `factor_y`)

Comment (ismael-mendoza) [reply jax_galsim/bounds.py:131]: The function works the same though so not very important

**Review by ismael-mendoza — APPROVED** [review]
LGTM! Just one minor comment

File: jax_galsim/bounds.py (hunk 2 above)
Comment (beckermr) [reply jax_galsim/bounds.py]: fixed!

**Pull request merged**